api/src/main/java/jakarta/data/spi/provider/DataProvider.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (c) 2023 Contributors to the Eclipse Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ */
+package jakarta.data.spi.provider;
+
+import java.lang.annotation.Annotation;
+import java.util.Set;
+
+import jakarta.data.exceptions.MappingException;
+
+/**
+ * <p>Jakarta Data providers can make themselves discoverable by
+ * Jakarta EE products via the Jakarta Data Provider SPI.
+ * Jakarta EE products with CDI support are thus able to automatically provide
+ * a CDI extension on behalf of the discovered Jakarta Data providers
+ * without those Jakarta Data providers themselves needing to depend on CDI.
+ * Jakarta Data providers with their own custom dependency injection
```

Comment:
```suggestion
 * Jakarta Data providers with dependency injection
```
IMHO: it only makes sense if you don't have any DI at all.
But, I may be missing something.